### PR TITLE
BNH13

### PR DESCRIPTION
**Pull request by SleepyUser**
Added Error pages for a range of status codes. See ErrorService file for full list.

ErrorView returns returns the Code, and a Name and Message associated with that code (if it is listed, otherwise a generic message is displayed).

Formatting and design of the error page still required in future to bring in line with overall visual style of site.

### CODE REVIEW

File: web/Program.cs [hunk 1]
```diff
@@ -47,9 +47,11 @@
     app.UseExceptionHandler("/Home/Error");
```
Comment (BenRamchandani) [inline web/Program.cs:47]: Should change this to "/Home/Error/500"

**Review by BenRamchandani — COMMENTED** [review]
This approach seems sensible to me. If you're happy with it tidy it up a bit and then ping me again and I can have another look.

File: web/ViewModels/ErrorViewModel.cs [hunk 2]
```diff
@@ -5,4 +5,8 @@ public class ErrorViewModel
     public string? RequestId { get; set; }
 
     public bool ShowRequestId => !string.IsNullOrEmpty(RequestId);
+
+    public string? StatusName { get; set; }
```
Comment (emilypatsko) [inline web/ViewModels/ErrorViewModel.cs:9]: Do these need to be nullable?

File: web/Services/ErrorService.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
+﻿namespace BricksAndHearts.Services;
+public class ErrorService
```
[the remark this answers is not in the document]
Comment (BenRamchandani) [reply web/Services/ErrorService.cs:2]: If this is just going to have a static method I'd be fine for it not to have an interface, but we shouldn't call it service, could call it e.g. `ErrorCodeMapping`

File: web/Views/Shared/Error.cshtml [hunk 4]
```diff
@@ -9,11 +9,14 @@
...
-        <strong>Request ID:</strong> <code>@Model.RequestId</code>
+        <strong>Error Code: @Model.RequestId</strong> <code>@Model.StatusName</code>
+    </p>
+    <p>
+        <code>@Model.StatusMessage</code>
     </p>
 }
 
-<h3>Development Mode</h3>
+@*<h3>Development Mode</h3>
```
Comment (BenRamchandani) [inline web/Views/Shared/Error.cshtml:19]: Let's delete this, we don't need the commented out stuff

File: web/Services/ErrorService.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (cec200wire) [reply web/Services/ErrorService.cs:2]: We changed it to use an interface.

File: web/Services/ErrorService.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
+﻿namespace BricksAndHearts.Services;
+
+public interface IErrorService
+{
+    public (string statusName, string statusMessage) GetStatusMessage(int statusCode);
+}
+public class ErrorService : IErrorService
+{ 
+    public (string statusName, string statusMessage) GetStatusMessage(int statusCode)
```
Comment (emilypatsko) [inline web/Services/ErrorService.cs:9]: I think these should be in pascal case, i.e. `StatusName` not `statusName`, pedantic I know 😄

**Review by emilypatsko — DISMISSED** [review]
I've just asked for two very small changes in `ErrorService`, please make these but then go ahead and merge :)

**Review by emilypatsko — APPROVED** [review]

**Review by emilypatsko — APPROVED** [review]

**Pull request merged**